### PR TITLE
Frontend access control

### PR DESCRIPTION
**Pull request by mdulin2**
Huge PR!  
- In particular, @jacobkrantz I would like for you to check the routes. At the moment, the add user does not have auth because it is convenient to be able to create users from the terminal when it is needed. 
- @andrewjyang please check for styling and such.

### CODE REVIEW

File: WebPhapp/backend/server.js [hunk 1]
```diff
@@ -574,23 +666,32 @@ app.post('/api/v1/users/add', auth.checkAuth([]), (req,res) => {
...
-        mysql.insertSalt(id, salt, connection)
-        .then( () => {
-            res.status(200).send(true);
-        });
-    })
-    .catch((error) => {
+        return mysql.insertSalt(id, salt, connection);
+
+    }).then(() => {
+        console.log("User created of role", userInfo.role, );
```
Comment (jacobkrantz) [inline WebPhapp/backend/server.js]: Preface with `/api/v1/users/add:`

File: WebPhapp/backend/mysql_helper.js [hunk 2]
```diff
@@ -112,6 +112,26 @@ module.exports = {
...
+        SET R.id_number = (R.id_number + 1)
+        WHERE R.role = ?;
+
+        SELECT id_number
+        FROM Role_Id_Count
+        WHERE role = ?;
+        `;
+
+        return new Promise((resolve, reject) => {
+            var values = [role, role];
```
[the remark this answers is not in the document]
Comment (mdulin2) [reply WebPhapp/backend/mysql_helper.js]: It is quite weird, but there is a reason for it, yes.

File: WebPhapp/backend/server.js (hunk 1 above)
Comment (mdulin2) [reply WebPhapp/backend/server.js]: Okay, thanks for that.

File: WebPhapp/backend/mysql_helper.js [hunk 3]
```diff
@@ -112,6 +112,26 @@ module.exports = {
         });
     },
 
+    updateRoleCount: function(role, connection){
```
[the remark this answers is not in the document]
Comment (mdulin2) [reply WebPhapp/backend/mysql_helper.js]: Whoops. Thanks for pointing that out!

File: README.md [hunk 4]
```diff
@@ -121,3 +121,19 @@ cd WebPhapp/
...
+```
+error: TypeError: values.5.map is not a function
+```
+make sure you are running your backend with npm run start
+not npm start.
+
+**Default Users**
+There are several users that are created in the DB. Here are the credentials for testing:
+  Username-Password-role-role_id
+- jackson oz: Prescriber 1
```
[the remark this answers is not in the document]
Comment (mdulin2) [reply README.md]: Agreed.

**Comment by mdulin2** [comment]
> Shablam! And the backend tests work! I see you are merging to `Access_Control`. This has got to be close to ready for `master`. What else needs to happen before we merge to `master`?

@jacobkrantz  I wanted to have all of the pages with the proper permissions up until dispensers. So, this weekend, Andrew and I will add that in. Then, it will be added.   

On a side note, I also need to update the connections.js file too.

**Comment by mdulin2** [comment]
@jacobkrantz I made the changes; let me know if there is anything else to do.

**Comment by andrewjyang** [comment]
Included in frontend access control pt. 2 pr

**Pull request closed without merging**